executor.go

```diff
@@ -27,14 +27,16 @@ const (
 type executor struct {
 	jobFunctions   chan jobFunction
 	stopCh         chan struct{}
+	stoppedCh      chan struct{}
 	limitMode      limitMode
 	maxRunningJobs *semaphore.Weighted
 }
 
 func newExecutor() executor {
 	return executor{
 		jobFunctions: make(chan jobFunction, 1),
-		stopCh:       make(chan struct{}, 1),
+		stopCh:       make(chan struct{}),
```

Comment:
We don't need capacity for this channel. Closing the channel is enough to tell the stop.